### PR TITLE
Make nft.trades Trino compatible again

### PR DESCRIPTION
**Pull request by belen-pruvost**

### CODE REVIEW

**Comment by belen-pruvost** [comment]
Regarding this PR - I ran `dbt run --select +nft_trades` locally, without errors.

I then checked whether `dbt_belen_nft.trades` was queryable in dune.com ([query link](https://dune.com/queries/1716405?d=1)) and everything is working (on the data explorer in Databricks I can also see that the table was created today). 

The only _issue_ I have is that these tables are created with dbfs locations, which means I can't query them on Trino. @jeff-dude do you have any idea on a workaround for that?

File: models/opensea/opensea_trades.sql [hunk 1]
```diff
@@ -47,7 +47,7 @@ CAST(NULL AS VARCHAR(5)) as collection,
 amount_usd,
 token_standard,
 CAST(NULL AS VARCHAR(5)) as trade_type,
-CAST(NULL AS DECIMAL(38,10)) as number_of_items,
+CAST(NULL AS DECIMAL(38,0)) as number_of_items,
```
[the remark this answers is not in the document]
Comment (belen-pruvost) [reply models/opensea/opensea_trades.sql:50]: Actually, this is a null being casted, not a field taken from another view underneath, so I believe this cast should stay here

[the remark this answers is not in the document]
Comment (jeff-dude) [reply models/opensea/opensea_trades.sql:50]: hmm, okay, i guess that query has a ton of cast statements. ideally in the future we clean that up to be at the base level, but we'll leave that for more general nft cleanup

**Comment by jeff-dude** [comment]
> Regarding this PR - I ran `dbt run --select +nft_trades` locally, without errors.
> 
> I then checked whether `dbt_belen_nft.trades` was queryable in dune.com ([query link](https://dune.com/queries/1716405?d=1)) and everything is working (on the data explorer in Databricks I can also see that the table was created today).
> 
> The only _issue_ I have is that these tables are created with dbfs locations, which means I can't query them on Trino. @jeff-dude do you have any idea on a workaround for that?

which table is giving you the dbfs issue? [this pr](https://github.com/duneanalytics/spellbook/pull/1867) introduced the creation of incremental tables to s3

**Comment by jeff-dude** [comment]
> > Regarding this PR - I ran `dbt run --select +nft_trades` locally, without errors.
> > I then checked whether `dbt_belen_nft.trades` was queryable in dune.com ([query link](https://dune.com/queries/1716405?d=1)) and everything is working (on the data explorer in Databricks I can also see that the table was created today).
> > The only _issue_ I have is that these tables are created with dbfs locations, which means I can't query them on Trino. @jeff-dude do you have any idea on a workaround for that?
> 
> which table is giving you the dbfs issue? [this pr](https://github.com/duneanalytics/spellbook/pull/1867) introduced the creation of incremental tables to s3

if you're referring to tables created in your local setup, i think that is normal (mine does this too) -- we'd need to setup an env variable to get this macro applied and write to s3.

**Comment by belen-pruvost** [comment]
> > Regarding this PR - I ran `dbt run --select +nft_trades` locally, without errors.
> > I then checked whether `dbt_belen_nft.trades` was queryable in dune.com ([query link](https://dune.com/queries/1716405?d=1)) and everything is working (on the data explorer in Databricks I can also see that the table was created today).
> > The only _issue_ I have is that these tables are created with dbfs locations, which means I can't query them on Trino. @jeff-dude do you have any idea on a workaround for that?
> 
> which table is giving you the dbfs issue? [this pr](https://github.com/duneanalytics/spellbook/pull/1867) introduced the creation of incremental tables to s3

For example, `dbt_belen_opensea_v1_ethereum.events` (link [here](https://dbc-7ee55e77-eabe.cloud.databricks.com/sql/explore/data/hive_metastore/dbt_belen_opensea_v1_ethereum/events?o=4190327660257548))

**Comment by belen-pruvost** [comment]
> if you're referring to tables created in your local setup, i think that is normal (mine does this too) -- we'd need to setup an env variable to get this macro applied and write to s3.

Yes, I'm referring to that. The problem with this is that I can't test them on Trino. But that's okay, I can iterate on the change if needed (as long as I don't break Databricks)

**Comment by jeff-dude** [comment]
> > if you're referring to tables created in your local setup, i think that is normal (mine does this too) -- we'd need to setup an env variable to get this macro applied and write to s3.
> 
> Yes, I'm referring to that. The problem with this is that I can't test them on Trino. But that's okay, I can iterate on the change if needed (as long as I don't break Databricks)

fair. we can both look to get our local envs fixed for this outside of gh. you're right, most important is to not break databricks with merging of PRs, so i think we can proceed and iterate.

**Comment by jeff-dude** [comment]
do we want to merge this now and see how it goes in prod, or feel better waiting for tomorrow morning?

**Comment by belen-pruvost** [comment]
If it's fine for you, let's merge. I'll keep an eye on the job, if it fails I'll revert these changes ASAP (will need you for PR approval on that though 😅 ).

**Comment by jeff-dude** [comment]
sure, we can go ahead, just wanted to ensure you'd be around if we broke `nft.trades` again 🥲

**Pull request merged**